service/task.go

```diff
@@ -889,3 +874,76 @@ func (uis *UIServer) testLog(w http.ResponseWriter, r *http.Request) {
 		uis.render.WriteResponse(w, http.StatusOK, data, "base", template)
 	}
 }
+
+func readBuildloggerToSlice(taskID string, r io.ReadCloser) []apimodels.BuildloggerLogLine {
+	lines := []apimodels.BuildloggerLogLine{}
+	lineChan := make(chan apimodels.BuildloggerLogLine, 1024)
+	go readBuildloggerToChan(taskID, r, lineChan)
+
+	for {
+		line, more := <-lineChan
+		if !more {
+			break
+		}
+
+		lines = append(lines, line)
+	}
+
+	return lines
+}
+
+func readBuildloggerToChan(taskID string, r io.ReadCloser, lines chan apimodels.BuildloggerLogLine) {
+	var (
```

Comment:
I'd feel better if this took a context and returned a receive only channel

Reply:
you mean send only, right?